src/analyze/analyze-chid.c

```diff
@@ -223,6 +247,47 @@ static int smbios_fields_acquire(char16_t *fields[static _CHID_SMBIOS_FIELDS_MAX
         return 0;
 }
 
+static int edid_parse(const char *edid_path, char16_t **ret_panel) {
+        int r;
+        _cleanup_free_ char *edid_content = NULL;
+        size_t edid_size = 0;
+
+        r = read_full_file(edid_path, &edid_content, &edid_size);
+        if (r < 0)
+                return r;
+
+        EdidHeader header;
+        if (!edid_parse_blob(edid_content, edid_size, &header))
+                return -EINVAL;
+
+        EdidPanelId panel_id;
```

Comment:
honestly, i am not really convinced we need to EdidPanelId structure in between. just do edid_get_panel_id() as before, that takes the edidheader structure, and generates a proper, valid utf16 string, by encoding the product id in hex.